isisd/isis_spf_delay.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * This is an implementation of the IETF SPF delay algorithm
+ * as explain in draft-ietf-rtgwg-backoff-algo
+ *
+ * Module name: isis_spf_delay.h 
+ * Version:     0.1
+ * Created:     25-01-2017 by S. Litkowski
+ * Copyright (C) 2017 Orange Labs http://www.orange.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston,
+ * MA 02110-1301 USA
+ */
+
+#ifndef ISIS_SPF_DELAY_H
+#define ISIS_SPF_DELAY_H
+
+#define	ISIS_SPF_DELAY_STATE_QUIET				0
+#define ISIS_SPF_DELAY_STATE_SHORT_WAIT			1
+#define ISIS_SPF_DELAY_STATE_LONG_WAIT			2
+
+extern const char	*isis_spf_delay_states[3];
+	
+struct isis_spf_delay_reinit {
+  // Struct used to store some informations needed to reinit SPF algo
+  struct isis_area	*area;
+  int			level;
+  int			family;
+};
+
+struct isis_spf_delay_ietf_af {
+  struct timeval	first_event_time;	// Store first event timestamp msec
+  struct timeval	last_event_time;	// Store last event received timestamp msec
+  struct timeval	next_spf;		// Store timestamp (msec) of next spf
+  struct timeval	resettime;		// Store timestamp of holddown firing
+  struct timeval	ttl_time;		// Store timestamp of timetolearn firing	
+  unsigned char         state;			// Store state of algo
+  struct thread         *t_spf;			// Thread for SPF schedule timer
+  int		        pending;		// Store 1 if SPF is pending, 0 instead	
+  struct thread         *t_holddown;		// Thread for HOLDDOWN timer
+  struct thread	        *t_timetolearn;		// Thread for TIMETOLEARN timer
+	
+};
+
+
+
+struct isis_spf_delay_ietf
+{
+	unsigned int	short_delay;
+	unsigned int	init_delay;
+	unsigned int	long_delay;
+	unsigned int	holddown;
+	unsigned int	timetolearn;
+	struct isis_spf_delay_ietf_af	*family[2];	// One struct per AF (AF_INET and AF_INET6)
```

Comment:
I'm not a fan of using a hard coded value of 2( even with the documentation here)  bgp uses AFI_MAX and the appropriate afi_t enum and an appropriate conversion function to specify the right spot (See bgpd.h 'struct bgp' there are several examples)  in the array.  Why wouldn't we want to use the same approach for this code?